packages/runtime/src/runtimes/react/utils/can-accept-ref.ts

```diff
@@ -0,0 +1,56 @@
+import type { Component, PropsWithoutRef, RefAttributes } from 'react'
+import { type ComponentType } from '../../../state/react-page'
+
+type WrapperComponent = {
+  $$typeof: string
+  new (props: PropsWithoutRef<any> & RefAttributes<any>, context?: any): Component<any>
+}
+
+type MemoComponent = WrapperComponent & {
+  type: ComponentType
+}
+
+// See https://github.com/facebook/react/blob/main/packages/shared/ReactSymbols.js
+const REACT_MEMO_TYPE: symbol = Symbol.for('react.memo')
+const REACT_LAZY_TYPE: symbol = Symbol.for('react.lazy')
+const REACT_FORWARD_REF_TYPE: symbol = Symbol.for('react.forward_ref')
```

Comment:
https://github.com/facebook/react/blob/main/packages/shared/ReactSymbols.js

Reply:
`Symbol.for` creates a [shared symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol#shared_symbols_in_the_global_symbol_registry):

> Because registered symbols can be arbitrarily created anywhere, they behave almost exactly like the strings they wrap.